clients/src/main/java/org/apache/kafka/common/protocol/types/Struct.java

```diff
@@ -290,6 +290,7 @@ public ByteBuffer getBytes(String name) {
         ByteBuffer buf = (ByteBuffer) result;
         byte[] arr = new byte[buf.remaining()];
         buf.get(arr);
+        buf.flip();
```

Comment:
Nice find.

Reply:
On the other hand I was debating whether it's worth to use `buf.duplicate().get()` instead of flipping on the expense of creating a lightweight ByteBuffer object. Advantages is that we don't mutate the current buffer but instead work on a throwaway object.

Reply:
Yeah, I too was unsure which way to go. I'm happy with either solutions, let's see what a committer thinks

Reply:
@mimaison Is this causing any issue?  If not, Can we just add javadoc saying "Reads the remaining bytes in a ByteBuffer into a byte[]."  and retain current code.

cc @ijuma

Reply:
By not flipping the buffer, the position stays at the end. Then when comparing Requests, the ones built via a Struct were not equals to the ones built directly.